py/tests/integration/common.py

```diff
@@ -220,26 +223,22 @@ def is_tx_confirmed(ext_api, tx_hash, min_confirmations):
 def get_account_balance(api, pub_key):
     return _balance_from_get_account(lambda: api.get_account_by_pubkey(pub_key), pub_key)
 
-def send_tokens_to_unchanging_user(sender, address, tokens, fee, external_api, internal_api):
+def send_tokens(sender, address, tokens, fee, ext_api, int_api):
     spend_tx_obj = SpendTx(
         sender_id=sender['enc_pubk'],
         recipient_id=address,
         amount=tokens,
         fee=fee,
         ttl=100,
         payload="sending tokens")
-    spend_tx = internal_api.post_spend(spend_tx_obj).tx
+    spend_tx = int_api.post_spend(spend_tx_obj).tx
     unsigned_tx = api_decode(spend_tx)
     signed_tx = integration.keys.sign_encode_tx(unsigned_tx, sender['privk'])
-    external_api.post_transaction(Tx(tx=signed_tx))
-
-def send_tokens_to_unchanging_user_and_wait_balance(sender, address, tokens, fee, ext_api, int_api):
-    bal0 = get_account_balance(ext_api, address)
-    send_tokens_to_unchanging_user(sender, address, tokens, fee, ext_api, int_api)
-    top = ext_api.get_current_key_block()
-    wait_until_height(ext_api, top.height+1)
-    wait(lambda: get_account_balance(ext_api, address) == (bal0 + tokens),
-         timeout_seconds=20, sleep_seconds=0.25)
+    return post_transaction(ext_api, signed_tx)
+
+def ensure_send_tokens(sender, address, tokens, fee, ext_api, int_api, min_confirmations):
```

Comment:
`is_tx_confirmed` waits until tx is in a microblock MB **and** the top key block has height that is at least `min_confirmations` greater than the height of MB.

If `min_confirmations` is 1 or greater, this guarantees that a node mined at least a key block on the microblock containing the tx. If the test has only one node, this guarantees avoiding micro-forks. The tests I checked have only one node. For tests having more than one node, in the future we may have to increase the number of confirmations so to avoid micro-forks (a node mining 2 key blocks without tx while a node mined 1 key block with tx).